### PR TITLE
Rename apiVersion YAML field to api_version

### PR DESCRIPTION
**Pull request by drevell**
This is the only field that's camel case, and it should be snake case for consistency with the other fields. It was a mistake in the first place for it to be capitalized differently (oops 🫤).

For backward compatibility, we'll still accept `apiVersion`, but we'll prefer `api_version`.

### CODE REVIEW

**Review by chloechien — DISMISSED** [review]

**Review by sethvargo — DISMISSED** [review]
I dig the backwards-compat! Does it make sense to print a warning for the old format?

**Comment by drevell** [comment]
> I dig the backwards-compat! Does it make sense to print a warning for the old format?

That's a good idea. I wonder if our YAML library makes that hard to do cleanly though. We don't have access to a `context.Context` in UnmarshalYAML(). So we'd have to either resort to atrocities like plumbing a context.Context through a global variable or constructing a new logger without the right logLevel. The cure seems worse than the disease here. But maybe there's a way to do it cleanly that I'm missing.

File: templates/model/spec_test.go [hunk 1]
```diff
@@ -72,6 +72,64 @@ steps:
 				},
 			},
 		},
+		{
+			name: "apiVersion_camel_case",
+			in: `api_version: 'cli.abcxyz.dev/v1alpha1'
```
[the remark this answers is not in the document]
Comment (drevell) [reply templates/model/spec_test.go]: Oof, fixed, thanks for catching that.

**Review by yolocs — APPROVED** [review]

**Comment by drevell** [comment]
> > I dig the backwards-compat! Does it make sense to print a warning for the old format?
> 
> That's a good idea. I wonder if our YAML library makes that hard to do cleanly though. We don't have access to a `context.Context` in UnmarshalYAML(). So we'd have to either resort to atrocities like plumbing a context.Context through a global variable or constructing a new logger without the right logLevel. The cure seems worse than the disease here. But maybe there's a way to do it cleanly that I'm missing.

Actually I have an idea how this could work that's a little sketchy. I'll send it as a followup PR since it may be controversial.

**Review by yolocs — APPROVED** [review]

**Comment by drevell** [comment]
I ended up partially rolling this back in #192 because it effectively caused an outage. When people used their existing abc CLI version, they were unable to install any of the templates that were hosted in github.com/abcxyz/abc because the old CLI didn't recognized the api_version field. I'm still thinking about the right way to handle breaking schema changes like this in the future.

**Pull request merged**